pkg/controller/bastion/options.go

```diff
@@ -0,0 +1,195 @@
+		return nil, err
+	}
+
+	region := cluster.Shoot.Spec.Region
+	return &Options{
+		Shoot:               cluster.Shoot,
+		BastionInstanceName: baseResourceName,
+		Zone:                getZone(cluster, region, providerStatus),
+		DiskName:            DiskResourceName(baseResourceName),
+		CIDRs:               cidrs,
+		Subnetwork:          fmt.Sprintf("regions/%s/subnetworks/%s", region, NodesResourceName(clusterName)),
+		ProjectID:           projectID,
+		Network:             fmt.Sprintf("projects/%s/global/networks/%s", projectID, clusterName),
+		WorkersCIDR:         workersCidr,
+	}, nil
+}
+
+func getZone(cluster *extensions.Cluster, region string, providerStatus *providerStatusRaw) string {
+	if providerStatus != nil {
+		return providerStatus.Zone
+	}
+
+	for _, j := range cluster.CloudProfile.Spec.Regions {
+		if j.Name == region {
+			if len(j.Zones) > 0 {
+				return j.Zones[0].Name
+			}
+		}
+	}
+	return ""
+}
+
+func ingressPermissions(bastion *extensionsv1alpha1.Bastion) ([]string, error) {
+	var cidrs []string
+	for _, ingress := range bastion.Spec.Ingress {
+		cidr := ingress.IPBlock.CIDR
+		ip, ipNet, err := net.ParseCIDR(cidr)
+		if err != nil {
+			return nil, fmt.Errorf("invalid ingress CIDR %q: %w", cidr, err)
+		}
+
+		normalisedCIDR := ipNet.String()
+
+		if ip.To4() != nil {
+			cidrs = append(cidrs, normalisedCIDR)
+		} else if ip.To16() != nil {
+			// Only IPv4 is supported in sourceRanges[].
+			// https://cloud.google.com/compute/docs/reference/rest/v1/firewalls/insert
+			return nil, fmt.Errorf("IPv6 is currently not fully supported: %w", err)
+		}
+
+	}
+
+	return cidrs, nil
+}
+
+func generateBastionBaseResourceName(clusterName string, bastionName string) (string, error) {
+	if clusterName == "" {
+		return "", fmt.Errorf("clusterName can't be empty")
+	}
+	if bastionName == "" {
+		return "", fmt.Errorf("bastionName can't be empty")
+	}
+
+	staticName := clusterName + "-" + bastionName
+	h := sha256.New()
+	_, err := h.Write([]byte(staticName))
+	if err != nil {
+		return "", err
+	}
+	hash := fmt.Sprintf("%x", h.Sum(nil))
+	if len([]rune(staticName)) > maxLengthForBaseName {
+		staticName = staticName[:maxLengthForBaseName]
+	}
+	return fmt.Sprintf("%s-bastion-%s", staticName, hash[:5]), nil
+}
+
+func getProviderStatus(bastion *extensionsv1alpha1.Bastion) (*providerStatusRaw, error) {
+	if bastion.Status.ProviderStatus != nil && bastion.Status.ProviderStatus.Raw != nil {
+		return unmarshalProviderStatus(bastion.Status.ProviderStatus.Raw)
+	}
+	return nil, nil
+}
+
+func marshalProviderStatus(zone string) ([]byte, error) {
+	return json.Marshal(&providerStatusRaw{
+		Zone: zone,
+	})
+}
+
+func unmarshalProviderStatus(bytes []byte) (*providerStatusRaw, error) {
+	info := &providerStatusRaw{}
+
+	err := json.Unmarshal(bytes, info)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse json for status.ProviderStatus")
+	}
+	return info, nil
+}
+
+func DiskResourceName(baseName string) string {
```

Comment:
doc strings for exported functions missing

Reply:
fixed in https://github.com/gardener/gardener-extension-provider-gcp/pull/288/commits/33570f2898da5b814bb3aebcb5c8766c0f4be77a